modules/apps/headless/headless-builder/headless-builder-test/src/testIntegration/java/com/liferay/headless/builder/model/listener/test/APISortRelevantObjectEntryModelListenerTest.java

```diff
@@ -7,92 +7,238 @@
 
 import com.liferay.headless.builder.application.APIApplication;
 import com.liferay.headless.builder.test.BaseTestCase;
+import com.liferay.petra.string.StringBundler;
 import com.liferay.petra.string.StringPool;
 import com.liferay.portal.kernel.json.JSONObject;
 import com.liferay.portal.kernel.json.JSONUtil;
+import com.liferay.portal.kernel.test.rule.AggregateTestRule;
+import com.liferay.portal.kernel.test.rule.DataGuard;
 import com.liferay.portal.kernel.test.util.HTTPTestUtil;
 import com.liferay.portal.kernel.test.util.RandomTestUtil;
 import com.liferay.portal.kernel.util.Http;
 import com.liferay.portal.kernel.util.StringUtil;
 import com.liferay.portal.test.rule.FeatureFlags;
+import com.liferay.portal.test.rule.LiferayIntegrationTestRule;
 
-import org.junit.Assert;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Rule;
 import org.junit.Test;
 
+import org.skyscreamer.jsonassert.JSONAssert;
+import org.skyscreamer.jsonassert.JSONCompareMode;
+
 /**
  * @author Sergio Jiménez del Coso
  */
+@DataGuard(scope = DataGuard.Scope.METHOD)
 @FeatureFlags({"LPS-167253", "LPS-178642"})
 public class APISortRelevantObjectEntryModelListenerTest extends BaseTestCase {
 
+	@ClassRule
+	@Rule
+	public static final AggregateTestRule aggregateTestRule =
+		new LiferayIntegrationTestRule();
+
+	@Before
+	@Override
+	public void setUp() throws Exception {
+		super.setUp();
+
+		_objectDefinitionJSONObject = _addObjectDefinition();
+	}
+
 	@Test
-	public void test() throws Exception {
-		JSONObject apiApplicationJSONObject = HTTPTestUtil.invokeToJSONObject(
+	public void testPostSort() throws Exception {
+		_addAPIApplication(
+			_objectDefinitionJSONObject.getString("externalReferenceCode"));
+
+		JSONAssert.assertEquals(
 			JSONUtil.put(
-				"applicationStatus", "published"
+				"status", JSONUtil.put("code", 0)
+			).toString(),
+			HTTPTestUtil.invokeToJSONObject(
+				JSONUtil.put(
+					"objectFieldERC", RandomTestUtil.randomString()
+				).put(
+					"oDataSort", "test ne 1"
+				).put(
+					"r_apiEndpointToAPISorts_c_apiEndpointERC",
```

Comment:
I think that it is wrong, the name of the relationship that ends in ERC is not a setter property because it is a property that only indicates the information of the ERC of the related Object Definition. Try to use the `Id` one which is the one that is used to set the related Ids in the APIs